firebase-firestore/src/test/java/com/google/firebase/firestore/local/RemoteDocumentCacheTestCase.java

```diff
@@ -194,11 +195,27 @@ public void testDocumentsMatchingQuerySinceReadTime() {
 
     Query query = Query.atPath(path("b"));
     ImmutableSortedMap<DocumentKey, MutableDocument> results =
-        remoteDocumentCache.getAllDocumentsMatchingQuery(query, version(12));
+        remoteDocumentCache.getAllDocumentsMatchingQuery(query, IndexOffset.create(version(12)));
     List<MutableDocument> expected = asList(doc("b/new", 3, docData));
     assertEquals(expected, values(results));
   }
 
+  @Test
+  public void testDocumentsMatchingQuerySinceReadTimeAndDocumentKey() {
```

Comment:
optional: add test for nanoseconds range-exclusivity

Reply:
Would this be different from `testDocumentsMatchingQuerySinceReadTime()`? If so, can you provide a rough sketch?

Reply:
I was thinking you could modify the test to check for the nanosecond tiebreaker:
```java
@Test
public void testDocumentsMatchingQuerySinceReadTimeAndDocumentKey() {
  Map<String, Object> docData = map("data", 2);
  addTestDocumentAtPath("b/a", /* updateTime= */ 1, /* readTime= */ version(11,1));
  addTestDocumentAtPath("b/b", /* updateTime= */ 2, /*  readTime= = */ version(11,2));
  addTestDocumentAtPath("b/c", /* updateTime= */ 3, /*  readTime= = */ version(11,3));
  addTestDocumentAtPath("b/d", /* updateTime= */ 4, /*  readTime= = */ version(12,2));

  Query query = Query.atPath(path("b"));
  ImmutableSortedMap<DocumentKey, MutableDocument> results =
      remoteDocumentCache.getAllDocumentsMatchingQuery(
          query, IndexOffset.create(version(11,2)));
  List<MutableDocument> expected = asList(doc("b/c", 3, docData), doc("b/d", 4, docData));
  assertEquals(expected, values(results));
```

Reply:
Got it, that makes a lot of sense. Added one more test.